config/config.go

```diff
@@ -921,17 +922,32 @@ func fetchConfig(u *url.URL) ([]byte, error) {
 	}
 	req.Header.Add("Accept", "application/toml")
 	req.Header.Set("User-Agent", internal.ProductToken())
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, err
-	}
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to retrieve remote config: %s", resp.Status)
+	retries := 3
+	for i := 1; i <= retries; i++ {
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
```

Comment:
non-200 responses don't return errors, so we might want to not retry on err. err could indicate something more fundamental is wrong. Check the docs for what conditions Do returns errors.

Reply:
So it only returns on error either for client policy or failure to speak HTTP. So the use case was for when someones network is temporarily down I'm thinking the retries on http status code covers that so will just break if there is an err